api/envoy/config/rbac/v3/rbac.proto

```diff
@@ -338,6 +338,23 @@ message Principal {
     type.matcher.v3.StringMatcher principal_name = 2;
   }
 
+  message FilterStateIPMatcherConfig {
```

Comment:
Along similar lines, I think it would be better to add a pluggable matcher to `type.matcher.v3.FilterStateMatcher` rather than add this as a special case here.